### DISCUSSION TITLE
How to propose a new core rule without a related new ECMAScript feature?

### DISCUSSION THREAD

**Discussion by chdh** [eslint/eslint#16765] [Q&A] [answered]
How can I report this [problem](https://stackoverflow.com/questions/74978573/eslint-rule-to-detect-nested-assignments)?

I think it would require a new core rule to detect nested variable assignments. But it's not possible to write a GitHub issue for a new core rule without specifying a related new ECMAScript feature.

(My problem is similar to #14280, for which the answer was not helpful.)

**Comment by DMartens** [comment, marked as answer]
For this case you can use the rule [no-restricted-syntax](https://eslint.org/docs/latest/rules/no-restricted-syntax).
If I understand correctly you only want assignments as statements which can be achieved with the selector `AssignmentExpression[parent.type!="ExpressionStatement"]`.
[ESLint playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gICAgc2VsZWN0b3I6IFwiQXNzaWdubWVudEV4cHJlc3Npb25bcGFyZW50LnR5cGUhPSdFeHByZXNzaW9uU3RhdGVtZW50J11cIixcbiAgICBtZXNzYWdlOiBcIkRvIG5vdCBuZXN0IGFzc2lnbm1lbnRzXCJcbn1dICovXG5cbmZuKGEgPSBiKTtcbmEgJiYgKGIgPSBjKTtcbmEgPSBiID0gYzsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZSwiZXMyMDIxIjp0cnVlfX19)

**Reply by chdh** [reply]
Thanks, this solves the problem.
Would you like to write your answer also on StackOverflow?

**Reply by chdh** [reply]
How can I allow this?

```
for (let i = 1; i < 16; i *= 2) {
   console.log(i);
}
```

**Reply by mdjermanovic** [reply]
> for (let i = 1; i < 16; i *= 2) {
>    console.log(i);
> }

Try with this selector:

```
AssignmentExpression:not(ExpressionStatement > AssignmentExpression, ForStatement > AssignmentExpression.update)
```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gICAgc2VsZWN0b3I6IFwiQXNzaWdubWVudEV4cHJlc3Npb246bm90KEV4cHJlc3Npb25TdGF0ZW1lbnQgPiBBc3NpZ25tZW50RXhwcmVzc2lvbiwgRm9yU3RhdGVtZW50ID4gQXNzaWdubWVudEV4cHJlc3Npb24udXBkYXRlKVwiLFxuICAgIG1lc3NhZ2U6IFwiRG8gbm90IG5lc3QgYXNzaWdubWVudHNcIlxufV0gKi9cblxuYSA9IGI7IC8vIG9rXG5cbmZvciAobGV0IGkgPSAxOyBpIDwgMTY7IGkgKj0gMikgeyAvLyBva1xuICAgY29uc29sZS5sb2coaSk7XG59XG5cbmZuKGEgPSBiKTsgLy8gZXJyb3JcbmEgJiYgKGIgPSBjKTsgLy8gZXJyb3JcbmEgPSBiID0gYzsgLy8gZXJyb3IiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZSwiZXMyMDIxIjp0cnVlfX19)

**Reply by chdh** [reply]
Thanks. All my source code now passes the rule.
